runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -421,4 +423,39 @@ private void onTileObject(Tile tile, TileObject oldObject, TileObject newObject)
 			}
 		}
 	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (config.showShortcutLevel())
+		{
+			MenuEntry[] entries = client.getMenuEntries();
+
+			//Skip anything where there's 0 chance it's a agi shortcut.
+			if (event.getTarget() == null || event.getTarget() == ""){
+				return;
+			}
+
+			Map<TileObject,Obstacle> obstacles = getObstacles();
+			for (Obstacle o : obstacles.values())
+			{
+				//Compare the event ID to nearby obstacle ID's
+				for (int obId: o.getShortcut().getObstacleIds()){
```

Comment:
It could be worth refactoring AgilityShortcut to store `obstacleIds` as a `List` instead of an array so that this can be changed from `for (id : getObstacleIds()) { if (id == e.getIdentifier()) { /* ... */ } }` to become `if (o.getShortcut().getObstacleIds().contains(event.getIdentifier())`. This would have the added benefit of allowing this to become another guard clause instead of another giant nested block.

Reply:
Currently, I believe it to effectively be like this except shortcut is stored in a variable so I can get it's level requirement.

```
for (Obstacle nearbyObstacle : getObstacles().values())
		{
			AgilityShortcut shortcut = nearbyObstacle.getShortcut();
			if (Arrays.stream(shortcut.getObstacleIds()).anyMatch(i -> i == entryId))
```

Reply:
Streams are pretty inefficient to initialize, but since this is only run once per menu opening at most, I think that'll be fine.